knowit/utils.py

```diff
@@ -1,13 +1,52 @@
 # -*- coding: utf-8 -*-
 from __future__ import unicode_literals
 
+import os
+import sys
 from collections import OrderedDict
 from datetime import timedelta
 
+sys.path.insert(0, os.path.abspath('..'))
+
 import babelfish
-from six import string_types
+
+from six import PY2, string_types, text_type
+
 import yaml
 
+from knowit import VIDEO_EXTENSIONS
+
+
+def recurse_paths(paths):
+    """Return a filesystemencoded list of videofiles.
+
+    :param paths:
+    :type video_path: string or list
+    :return:
+    :rtype: list
+    """
+    enc_paths = []
+
+    if isinstance(paths, (string_types, text_type)):
+        paths = [paths]
+
+    encoding = sys.getfilesystemencoding()
+    for path in paths:
+        if os.path.isfile(path):
+            enc_paths.append(path.decode(encoding) if PY2 else path)
+        if os.path.isdir(path):
+            for root, directories, filenames in os.walk(path):
+                for filename in filenames:
+                    if os.path.splitext(filename)[1] in VIDEO_EXTENSIONS:
+                        if PY2 and os.name == 'nt':
+                            fullpath = os.path.join(root, filename.decode(encoding))
+                        else:
+                            fullpath = os.path.join(root, filename).decode(encoding)
+                        enc_paths.append(fullpath)
+
+    # Lets remove any dupes since knowit is rather slow.
+    return list(set(enc_paths))
```

Comment:
Getting rid of dupes is a really good idea :+1: 

But `list(set(enc_paths))` will not preserve the order.

Probably something like this might be enough:
```
    seen = set()
    seen_add = seen.add
    return [x for x in enc_paths if not (x in seen or seen_add(x))]
```